spec/features/user_integration_spec.rb

```diff
@@ -0,0 +1,98 @@
+require 'rails_helper'
+
+RSpec.feature 'User Integration', type: :feature do
+  let!(:user) { create(:user) }
+
+  scenario 'User index page' do
+    visit users_path
+
+    expect(page).to have_content(user.name)
+    expect(page).to have_selector('img.img-thumbnail')
+    expect(page).to have_content("Number of posts: #{user.posts_counter}")
+
+    click_link user.name
+    expect(current_path).to eq(user_path(user))
+  end
+
+  scenario 'User show page' do
+    visit user_path(user)
+
+    expect(page).to have_selector('img.img-thumbnail')
+    expect(page).to have_content(user.name)
+    expect(page).to have_content("Number of posts: #{user.posts_counter}")
+    expect(page).to have_content(user.bio)
+
+    if user.posts.any?
+      user.posts.limit(3).each do |post|
+        expect(page).to have_content(post.title)
+      end
+
+      expect(page).to have_link('See all Posts', href: user_posts_path(user))
+      click_link user.posts.first.title
+      expect(current_path).to eq(user_post_path(user, user.posts.first))
+    end
+  end
+
+  scenario 'User post index page' do
+    visit user_posts_path(user)
+
+    expect(page).to have_selector('img.img-thumbnail')
+    expect(page).to have_content(user.name)
+    expect(page).to have_content("Number of posts: #{user.posts_counter}")
+
+    user.recent_posts.each do |post|
+      expect(page).to have_content(post.title)
+      expect(page).to have_content(post.text.truncate(50))
+      expect(page).to have_content("Comments: #{post.comments_counter}")
+      expect(page).to have_content("Likes: #{post.likes_counter}")
+
+      click_link post.title
+      expect(current_path).to eq(user_post_path(user, post))
+      visit user_posts_path(user) # Go back to user post index
+    end
+  end
+
+  scenario 'Post show page' do
+    user = create(:user)
+    post = create(:post, author: user)
+
+    visit user_post_path(user, post)
+
+    expect(page).to have_content(post.title)
+    expect(page).to have_content("by #{post.author.name}")
+    expect(page).to have_content("Comments: #{post.comments_counter}")
+    expect(page).to have_content("Likes: #{post.likes_counter}")
+    expect(page).to have_content(post.text)
+
+    post.comments.each do |comment|
+      expect(page).to have_content(comment.user.name)
+      expect(page).to have_content(comment.text)
+    end
+  end
+
+  scenario 'Post show page displays first comments' do
+    post = create(:post, author: user)
+    comments = create_list(:comment, 3, post: post)
+
+    visit user_post_path(user, post)
+
+    expect(page).to have_content(post.title)
+    expect(page).to have_content("by #{post.author.name}")
+    expect(page).to have_content("Comments: #{post.comments_counter}")
+    expect(page).to have_content(post.text)
+
+    comments.each do |comment|
+      expect(page).to have_content(comment.user.name)
+      expect(page).to have_content(comment.text)
+    end
+  end
```

Comment:
- [ ] Great job so far making the changes suggested by the previous reviewer. However, One of your tests is still failing so kindly consider making the needed changes for it to pass. Below is the details of the error:
<img width="702" alt="Screenshot 2024-01-31 at 21 00 26" src="https://github.com/Eccb7/Pi-Blogs/assets/107192076/63b07695-eb52-4c37-995e-a4c4e57ff315">

Reply:
Tracing such from my end is hard coz here is my test result

![Screenshot from 2024-02-01 00-12-36](https://github.com/Eccb7/Pi-Blogs/assets/100193711/9b24ea12-d9ec-4f75-b6f7-5f54a51582c8)